env0/provider.go

```diff
@@ -155,26 +155,37 @@ func configureProvider(version string, p *schema.Provider) schema.ConfigureConte
 			isIntegrationTest = true
 		}
 
-		restyClient.
+		subCtx := tflog.NewSubsystem(ctx, "env0_api_client")
+
+		restyClient = restyClient.
 			SetRetryCount(5).
 			SetRetryWaitTime(time.Second).
 			SetRetryMaxWaitTime(time.Second * 5).
+			OnBeforeRequest(func(c *resty.Client, r *resty.Request) error {
+				tflog.SubsystemInfo(subCtx, "env0_api_client", "Sending request", map[string]interface{}{"method": r.Method, "url": r.URL})
+				return nil
+			}).
+			OnAfterResponse(func(c *resty.Client, r *resty.Response) error {
+				tflog.SubsystemInfo(subCtx, "env0_api_client", "Received respose", map[string]interface{}{"method": r.Request.Method, "url": r.Request.URL, "status": r.Status()})
+				return nil
+			}).
+			AddRetryAfterErrorCondition().
```

Comment:
This may have been needed to retry 4xx.
Not sure... but worth checking it out if it helps (documentation isn't very clear).

Reply:
looking in the code... it's just:
```func (c *Client) AddRetryAfterErrorCondition() *Client {
	c.AddRetryCondition(func(response *Response, err error) bool {
		return response.IsError()
	})
	return c
}```

but we'll try...